core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -246,13 +246,16 @@ bool thread_disable_prealloc_rpc_cache(uint64_t *cookie)
 	}
 
 	rv = true;
-	for (n = 0; n < CFG_NUM_THREADS; n++) {
-		if (threads[n].rpc_arg) {
-			*cookie = mobj_get_cookie(threads[n].rpc_mobj);
-			mobj_put(threads[n].rpc_mobj);
-			threads[n].rpc_arg = NULL;
-			threads[n].rpc_mobj = NULL;
-			goto out;
+
+	if (IS_ENABLED(CFG_PREALLOC_RPC_CACHE)) {
```

Comment:
In case this is true we can return true directly at the beginning of the function without any other checks or locks.

Reply:
Good point, I agree.